shared/src/error/mod.rs

```diff
@@ -1,101 +1,116 @@
-// Copyright 2019-2022 ChainSafe Systems
-// SPDX-License-Identifier: Apache-2.0, MIT
-
 use std::fmt::Formatter;
 
 use num_derive::FromPrimitive;
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
-use crate::encoding::repr::*;
-
-/// ExitCode defines the exit code from the VM execution.
-#[repr(u32)]
-#[derive(
-    PartialEq, Eq, Debug, Clone, Copy, FromPrimitive, Serialize_repr, Deserialize_repr, Error,
-)]
-pub enum ExitCode {
-    Ok = 0,
-
-    /// Indicates failure to find an actor in the state tree.
-    SysErrSenderInvalid = 1,
+/// ExitCode defines the exit code from the VM invocation.
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct ExitCode {
```

Comment:
We should probably implement `From` on this.

Reply:
That would let us replace all `ExitCode::new(...)` with `ExitCall::from(...)` (which is generally a bit more idiomatic).

Reply:
I implemented `From<u32>` but left `new` in place as I can't call `from()` from a `const` definition (and can't annotate it `const`).

Reply:
Note: it's common to use `pub struct ExitCode(u32)` in cases like this, although I don't have strong opinions here.